rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+
+
+## <a id="key"></a> b) Key Management and Addresses in Nescience
+
+The Nescience architecture utilizes difirrent cryptographic schemes, such as public key encryption and digital signatures, to ensure secure private executions through 
+the exchange of UTXOs. These schemes rely on a structured set of cryptographic keys, each serving a specific purpose in maintaining privacy, security, and control over assets. 
+Here's a breakdown of the keys used in Nescience:
+
+#### I. Spending Key
+The spending key is the fundamental secret key in the Nescience architecture, acting as the primary control mechanism for a user’s UTXOs and other digital assets. 
+It plays a critical role in the cryptographic security of the system, ensuring that only the rightful owner can authorize and spend their assets.
+
+- **Role of the Spending Key**: The spending key is responsible for generating the user’s private keys, which are used in various cryptographic operations such as 
+signing transactions and creating commitments. This hierarchical relationship means that the spending key sits at the root of a user’s key structure, safeguarding 
+access to all associated private keys and, consequently, to the user’s assets. In Nescience’s privacy-focused model, the spending key is never exposed or shared outside 
+the user’s control. Unlike other keys, it does not interact with the public state, kernel circuits, or even the zero-knowledge proof system. This isolation ensures that 
+the spending key remains completely private and inaccessible to external entities. By keeping the spending key separate from the operational aspects of the network, 
+Nescience minimizes the risk of key leakage or compromise.
+
+- **Generation and Security of the Spending Key**: The spending key is generated randomly from the scalar field, a large mathematical space that ensures uniqueness 
+and cryptographic strength. This randomness is crucial because it prevents attackers from predicting or replicating the key, thereby safeguarding the user’s assets 
+from unauthorized access: it is computationally infeasible for an attacker to guess or brute-force the key. Once the spending key is generated, it is securely stored 
+by the user, typically in a hardware wallet or another secure storage mechanism that prevents unauthorized access.
+
+- **Spending UTXOs with the Spending Key**: The spending key’s primary function is to authorize the spending of UTXOs in private transactions. When a user initiates 
+a transaction, the spending key is used to generate the necessary cryptographic proofs and signatures, ensuring that the transaction is valid and originates from 
+the rightful owner. However, even though the spending key generates these proofs, it is never directly exposed during the transaction process. Instead, derived 
+private keys handle the operational aspects, while the spending key remains securely in the background. For example, when Alice decides to spend a UTXO in a 
```

Comment:
remove comma between "aspects" and "while"
Consider changing "securely" to "secure"